### PR TITLE
Back End Trade Functionality

### PR DESCRIPTION
**Pull request by jacobstringer**
Adds in trade logic, and includes controller tests for the new trade sections - seems to be working properly. I have hooked it up to the process_card function and set it to automatically trade if the option is available, pending decisions around how the front end is set up for this. A boolean parameter 'trade' has been set up in the play_card API so we can hook something up to the front end.

AI is currently not using the trade function, as they bypass process_card and go straight to play_card. We can include them later, but I think until the AI has a bit of brains, adding them in for trading will be less than ideal.

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -266,8 +266,10 @@ def play_card():
     discarded = false_true(request.args.get('discarded'))
     for_wonder = false_true(request.args.get('for_wonder'))
     from_discard_pile = false_true(request.args.get('from_discard_pile'))
+    #trade = false_true(request.args.get('trade'))
```
Comment (mmdonaldson) [inline application/app.py]: Do we need this to be commented out? Maybe just add the default to the endpoints in the frontend

**Review by mmdonaldson — APPROVED** [review]
I see what you need about complicated stuff... Approved but if you wanted uncomment the trade args on the api and just add a default to the frontend

**Review by Sam-Irv — COMMENTED** [review]
Looks alright. Did you delete test_process_card_monetary_cost  for any particular reason though?

**Comment by jacobstringer** [comment]
@Sam-Irv, I didn't change any of the controller tests you created - just git being funny again.

@mmdonaldson, a default on the front end would be better. I just did it this way because I couldn't be bothered figuring out where to look at 1am lol.

**Pull request merged**